app/src/main/java/org/simple/clinic/patient/PatientRepository.kt

```diff
@@ -172,9 +173,41 @@ class PatientRepository @Inject constructor(
         }
   }
 
-  fun mergeWithLocalData(serverPayloads: List<PatientPayload>): Completable {
+  override fun recordsWithSyncStatus(syncStatus: SyncStatus): Single<List<PatientSaveModel>> {
+    return database.patientDao()
+        .recordsWithSyncStatus(syncStatus)
+        .firstOrError()
+        .map { results ->
+          results.asSequence()
```

Comment:
This is a little complicated, and I want to explain what's going on. 

`PatientSaveModel` has a non-nullable list of patient phone numbers and it expects that a Patient without phone numbers will have an empty list of phone numbers. However, `PatientQueryModel` is constructed from a join of three tables, and the phone number table is joined with a `LEFT JOIN` so missing phone numbers will be `null`.

What this does is checks if the first phone number for a Patient query model with a given patient ID is `null`, and if it is, sets the phone numbers of the `PatientSaveModel` to an empty list, otherwise it extracts all the phone numbers.

Reply:
> However, PatientQueryModel is constructed from a join of three tables, and the phone number table is joined with a LEFT JOIN so missing phone numbers will be null.

Can default values in Kotlin solve this problem?